src/wristpy/core/orchestrator.py

```diff
@@ -0,0 +1,259 @@
+    )
+
+    end_sequence = pl.repeat(
+        nonwear_upsample["nonwear"][-1],
+        n=(len(reference_measure.time) - len(nonwear_upsample["time"])),
+        dtype=pl.Int64,
+        eager=True,
+    )
+    full_nonwear_array = pl.concat([nonwear_upsample["nonwear"], end_sequence])
+
+    return full_nonwear_array.to_numpy()
+
+
+def run(
+    input: pathlib.Path,
+    output: Optional[pathlib.Path] = None,
+    settings: config.Settings = config.Settings(),
+    calibrator: calibration.Calibration = calibration.Calibration(),
+    detect_nonwear_kwargs: Optional[Dict[str, Any]] = None,
+) -> Results:
+    """Runs wristpy.
+
+    Args:
+        input: Path to the input file to be read. Currently supports .bin and .gt3x
+        output: Path to save data to. The path should end in the file name to be given
+            to the save data. Two files will be saved, each with the given file name and
+            the _raw_time or _epoch1 label after. Currently supports saving in .csv and
+            .parquet
+        settings: The settings object from which physical activity levels are taken.
+        calibrator: The calibrator to be used on the input data.
+        detect_nonwear_kwargs: The arguments to the nonwear function delivered as a
+            dictionary. Arguements are short_epoch_length, n_short_epoch_in_long_epoch,
+            std_criteria, range_criteria.
+
+    Returns:
+        All calculated data in a save ready format as a Results object.
+
+    """
+    watch_data = readers.read_watch_data(input)
+    try:
+        calibrated_acceleration = calibrator.run(watch_data.acceleration)
+    except Exception as e:
+        print("Calibration FAILED:", str(e))
+        print("Proceeding without calibration.")
+        calibrated_acceleration = watch_data.acceleration
+    enmo = metrics.euclidean_norm_minus_one(calibrated_acceleration)
+    enmo_epoch1 = computations.moving_mean(enmo)
+    anglez = metrics.angle_relative_to_horizontal(calibrated_acceleration)
+    anglez_epoch1 = computations.moving_mean(anglez)
+    non_wear_array = metrics.detect_nonwear(
+        calibrated_acceleration, **(detect_nonwear_kwargs or {})
+    )
+    sleep_detector = analytics.GGIRSleepDetection(anglez_epoch1)
+    sleep_windows = sleep_detector.run_sleep_detection()
+    physical_activity_levels = analytics.compute_physical_activty_categories(
+        enmo_epoch1,
+        (
+            settings.LIGHT_THRESHOLD,
+            settings.MODERATE_THRESHOLD,
+            settings.VIGOROUS_THRESHOLD,
+        ),
+    )
+    sleep_array = models.Measurement(
+        measurements=format_sleep_data(
+            sleep_windows=sleep_windows, reference_measure=enmo_epoch1
+        ),
+        time=enmo_epoch1.time,
+    )
+    nonwear_epoch1 = models.Measurement(
+        measurements=format_nonwear_data(
+            nonwear_data=non_wear_array, reference_measure=enmo_epoch1
+        ),
+        time=enmo_epoch1.time,
+    )
+
+    results = Results(
```

Comment:
Since creating epoch1 measures is always just a moving mean, could we not move this into the save_results step? i.e. somehting like `result.save_results(output, epoch=5)` or `result.save_results(output, epoch=None)`.

This would also make saving other epochs require no code duplication.